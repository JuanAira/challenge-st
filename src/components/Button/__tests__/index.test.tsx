import React from 'react';
import { render } from '@testing-library/react';
import ThemeProvider from '../../../context/Theme/Provider';
import { Button } from '..';

describe('Button component', () => {
  beforeAll(() => {
    window.matchMedia = window.matchMedia || function () {
      return {
        matches: false,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      };
    };
  });

  test('should match to snapshot - base', () => {
    const { container } = render(
      <ThemeProvider>
        <Button onClick={jest.fn()} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});