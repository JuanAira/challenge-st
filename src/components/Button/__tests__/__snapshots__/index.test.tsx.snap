// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match to snapshot - base 1`] = `
<div>
  <button
    class="sc-bczRLJ dlyGG"
  >
    <img
      alt="light theme hamburger"
      class="sc-gsnTZi ccUHbM"
      src="elements_hamburger_interface_list_menu_ui_icon_light.svg"
    />
  </button>
</div>
`;
